### PR TITLE
[EXCEL-MULTI-LOOKUP-5] PLU-593 (feat): support multiple filters

### PR DESCRIPTION
**Pull request by kevinkim-ogp**
### TL;DR

Update M365 Excel "Get Table Row", "Get Table Rows", and "Update Table Row" actions to support up to 3 lookup filters.

### What changed?

- The `maxRows` limit on the lookup conditions field for "Get Table Row", "Get Table Rows", and "Update Table Row" actions has been increased from 1 to 3, allowing users to specify up to 3 column/value filter conditions.
- `getTableRowImpl` and the "Get Table Rows" action now accept a `filters` array instead of a single `lookupColumn`/`lookupValue` pair. Rows are matched only when **all** filters match (AND logic).
- Column index resolution and row matching logic have been updated to iterate over all provided filters.
- Integration tests for "Get Table Rows" have been updated to use the `filters` array parameter and expanded to cover multi-filter scenarios (all match, partial match, empty string values).
- A new unit test file for `getTableRowImpl` covers single filter, multiple filters, and edge cases (empty rows, empty string values, non-zero header row index, case-sensitive matching).

### How to test?

- [ ] Verify that you can add up to 3 lookup conditions
    - [ ] Get table row
    - [ ] Get multiple table rows
    - [ ] Update table row
- [ ] Verify that the lookup conditions are applied correctly
    - [ ] Get table row should only return 1 row
    - [ ] Get multiple table rows
    - [ ] Update table row

### CODE REVIEW

**Comment by kevinkim-ogp** [comment]
* **#1584** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1584?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1583** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1583?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1583?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1582** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1582?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1581** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1581?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1601** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1601?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1600** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1600?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1599** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1599?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1578** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1578?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1579](https://github.com/opengovsg/plumber/pull/1579) <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1579?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1594** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1594?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1577** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1577?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1580** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1580?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `trunk/excel-multi-lookup`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by m0nggh — COMMENTED** [review]
nit to add strictier schema check:
oh actually should check for duplicate columns also
```
export const filtersSchema = z
  .array(
    z.object({
      lookupColumn: z.string().min(1),
      lookupValue: z.string().default(''),
    }),
  )
  .min(1, { message: 'Please add at least one lookup condition.' })
  .max(3, { message: 'Only up to 3 lookup conditions are supported.' })
  .refine(
    (filters) => {
      const columns = filters.map((f) => f.lookupColumn)
      return new Set(columns).size === columns.length
    },
    {
      message: 'Each lookup condition must use a different column.',
    },
  )
```

File: packages/backend/src/apps/m365-excel/actions/get-table-row/implementation/index.ts [hunk 1]
```diff
@@ -16,8 +19,7 @@ interface GetTableRowImplParams {
   $: IGlobalVariable
   session: WorkbookSession
   tableId: string
-  lookupValue: string
-  lookupColumn: string
+  filters: z.infer<typeof lookupParametersSchema>['filters']
```
Comment (m0nggh) [inline packages/backend/src/apps/m365-excel/actions/get-table-row/implementation/index.ts]:
```suggestion
  filters: z.infer<typeof filtersSchema>
```

**Review by m0nggh — APPROVED** [review]
other than the schema change, i think solid work leh, tested that old pipes with 1 condition still work, and new pipes with 1-3 conditions work as well

regarding the multiphase part, i think can do tgt ba, if really scared maybe consider a LD flag to change the `maxRows`? Can sense check with @pregnantboy on this baa 

I think quite safe to just deploy to staging/UAT, do the testing there with old pipes before deployment and monitor logs for any `transformLookupConditionsToFilters` errors, weird step parse failures, etc.

**Comment by kevinkim-ogp** [comment]
### Merge activity

* **May 25, 3:15 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/opengovsg/plumber/1583).
* **May 25, 3:16 AM UTC**: @kevinkim-ogp merged this pull request with [Graphite](https://app.graphite.com/github/pr/opengovsg/plumber/1583).
